### ISSUE TITLE
See notebook name in Browser Tab Title when notebook is inside nested folders

### ISSUE THREAD

**Issue by subhamagrawal7** [jupyterlab/jupyterlab#11463]
With Jupyter Lab 3, we now get the file name reflected directly in the title of the Browser. I found this feature very useful. But the issue is when the notebook is deep inside a folder structure, then the browser title does not reflect the notebook name correctly. 
For example, I have the following folder structure - 
```
$USER
   | experiments
       | notebooks
           | notebooks-test
              | my_first_ntbk.ipynb
              | my_second_ntbk.ipynb
              | my_third_ntbk.ipynb ... and so on
```

Now when I open my_first_ntbk.ipynb then the browser title shows this - 
![image](https://user-images.githubusercontent.com/34346812/141785024-4466f6c4-22e7-4d59-b128-2584dc9a856a.png)

If I could see the my_first_ntbk.ipynb directly then it would be very helpful.

Thanks,
Subham

**Comment by jtpio** [comment]
Thanks @subhamagrawal7 for the feedback.

It looks like this is the relevant code:

https://github.com/jupyterlab/jupyterlab/blob/7c8e29b242b4abfee240ec57262330ba62c75a0c/packages/apputils-extension/src/index.ts#L331-L365

And this seems to be fixed with the latest `4.0.0a14`:

![image](https://user-images.githubusercontent.com/591645/141790321-db2db14f-1705-440b-92f1-8b9767f55290.png)

Which version of JupyterLab 3 are you using? Maybe the corresponding change could then be backported to the 3.2.x release series if not already.

**Comment by subhamagrawal7** [comment]
> Which version of JupyterLab 3 are you using?

Using 3.2.3 currently

**Comment by jtpio** [comment]
@subhamagrawal7 would you be able to try with the latest 4.0 alpha release?

`pip install --pre jupyterlab`

Thanks!

**Comment by subhamagrawal7** [comment]
Yes with the latest 4.0 alpha release it is working as expected. 
I installed using `pip install jupyterlab==4.0.0a14`

**Comment by jtpio** [comment]
Thanks for checking.

The fix seems to have been added in https://github.com/jupyterlab/jupyterlab/pull/10952

It's tagged as 3.1.x but doesn't seem to be part of the `3.2.x` branch.

**Comment by jtpio** [comment]
Closing as fixed by #11465, which should be available in the next `3.2.x` release.

@subhamagrawal7 in the meantime you can use this Binder link to try it out:

https://mybinder.org/v2/gh/jupyterlab/jupyterlab/3.2.x?urlpath=lab-dev
